loading/lazyload/move-element-and-scroll.tentative.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<head>
+  <title>Images with loading='lazy' load being moved to another document and then scrolled to.</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="common.js"></script>
+</head>
+
+<!--
+Marked as tentative until https://github.com/whatwg/html/pull/3752 is landed.
+-->
+
+<body>
+  <img id="in_viewport" src='resources/image.png'>
+  <div style="height:10000px;"></div>
+  <img id="below_viewport" src='resources/image.png' loading="lazy">
+
+  <script>
+    var in_viewport_element = document.getElementById("in_viewport");
+    var below_viewport_element = document.getElementById("below_viewport");
+
+    async_test(function(t) {
+      var iframe = document.createElement('iframe');
+      iframe.srcdoc = "<body></body>";
+      iframe.onload = function() {
+        var adopted_img = iframe.contentDocument.adoptNode(below_viewport_element);
+        iframe.contentDocument.body.appendChild(adopted_img);
+        window.scrollTo(0, 10000);
+        t.step_timeout(function() { assert_false(is_image_fully_loaded(below_viewport_element)); t.done(); }, 1000);
+      };
+      document.body.insertBefore(iframe, in_viewport_element,);
+    }, "Test that <img> below viewport is not loaded when moved to another document and then scrolled to.");
```

Comment:
Just want to be clear on what this is testing exactly: it seems you're testing that when you remove a node and insert it into another document, and scroll to _the old place it used to exist in the DOM_, it will not load. Is this right?

Reply:
That is correct. However I identified a problem with the test. In WebKit implementation we have the rule that image elements moved between documents will be seen as "created from Javascript" and will not do lazy loading but eager loading instead. I think this is according to spec? In that case, is_image_fully_loaded will not do the right thing since it assumes lazy loading.
Let me know if you can think of a better way to test this and in general if you can think of similar edge cases let me know as well.

Reply:
One question: It seems that the tests's usage of document.body.insertBefore inserts the iframe (and therefore the image) into a portion of the DOM that intersects with the viewport. Won't this make the image start to load?

> However I identified a problem [...]

Ah I see, thanks for bringing this up. The spec actually uses the DOM's notion of connected-ness (exposed via [`.isConnected`](https://dom.spec.whatwg.org/#dom-node-isconnected) FWIW) to determine whether an image will be lazy loaded. This is something Maciej brought up doing in spec discussion and it seemed to work well; see https://whatpr.org/html/3752/3bbb870...034b96e/images.html#updating-the-image-data for more info. So just to clarify, by spec:
 - An image created from JavaScript & disconnected with no `loading` attribute, or `loading=auto`, will load eagerly
 - An image created from JavaScript & disconnected with `loading=lazy` will load lazily
 - The same image in the above point will then start loading if connected to a part of the DOM that intersects with the viewport.

So _I believe_ the test should be OK, and what you spotted is an (albeit minor) WebKit quirk?